.github/workflows/ci.yml

```diff
@@ -0,0 +1,90 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '**'
+  pull_request: {}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: dtolnay/rust-toolchain@stable
+      with:
+        components: rustfmt, clippy
+
+    - id: cache-rust
+      uses: Swatinem/rust-cache@v2
+
+    - uses: pre-commit/action@v3.0.0
+      with:
+        extra_args: --all-files --verbose
+      env:
+        PRE_COMMIT_COLOR: always
+        SKIP: test
+
+  test:
+    name: test rust-${{ matrix.rust-version }}
+    strategy:
+      fail-fast: false
+      matrix:
+        rust-version: [stable, nightly]
+
+    runs-on: ubuntu-latest
+
+    env:
+      RUST_VERSION: ${{ matrix.rust-version }}
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: ${{ matrix.rust-version }}
+
+    - id: cache-rust
+      uses: Swatinem/rust-cache@v2
+
+    - run: cargo test --all-features
+#    - uses: taiki-e/install-action@cargo-llvm-cov
```

Comment:
@alamb this is currently [not working](https://github.com/datafusion-contrib/datafusion-functions-json/actions/runs/8782026446/job/24095272341) as `datafusion-functions-json` is inactive on codecov, please can you enable it.

Reply:
Hi @samuelcolvin  -- I am not sure what you want me to do here. How do I enable code-cov?

Reply:
I assume you go to https://app.codecov.io/github/datafusion-contrib and there's a button to enable this repo.

Reply:
Done

I added the required repository secret (I think) as `CODECOV_TOKEN`

The instructions on https://app.codecov.io/github/datafusion-contrib/datafusion-functions-json/new now say

> Step 2: add Codecov to your [GitHub Actions workflow yaml file](https://github.com/datafusion-contrib/datafusion-functions-json/tree/main/.github/workflows)
> After tests run, this will upload your coverage report to Codecov:


```yml
- name: Upload coverage reports to Codecov
  uses: codecov/codecov-action@v4.0.1
  with:
    token: ${{ secrets.CODECOV_TOKEN }}
    slug: datafusion-contrib/datafusion-functions-json
```